moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -665,20 +696,21 @@ void JogCalcs::enforceSRDFAccelVelLimits(Eigen::ArrayXd& delta_theta)
       }
 
       // Apply velocity bounds
-      if (clip_velocity)
+      if (clip_velocity && delta_theta(joint_delta_index) != 0)
       {
         // delta_theta = joint_velocity * delta_t
         const double relative_change = (velocity_limit * parameters_.publish_period) / delta_theta(joint_delta_index);
-        // Avoid nan
-        if (fabs(relative_change) < 1)
+        if (relative_change < most_extreme_relative_change)
         {
-          delta_theta(joint_delta_index) = relative_change * delta_theta(joint_delta_index);
-          velocity(joint_delta_index) = relative_change * velocity(joint_delta_index);
+          most_extreme_relative_change = relative_change;
         }
       }
       ++joint_delta_index;
     }
   }
+
+  // Apply the limits
+  delta_theta *= most_extreme_relative_change;
```

Comment:
At first glance, this is a really good way of doing it.

But, what if a joint velocity is very large and most_extreme_relative_change is very small? That could cause a sudden deceleration, I think.

For example, say J1 caused `most_extreme_relative_change` = 0.1. J2 was previously at a velocity of 10 rad/s, now it will slow down to 1 rad/s. Now J2 might be violating an acceleration limit, right?

Reply:
Huh, I did not think about that. In this case if J2 is violating its limit afterward, then there is no scaling factor that can make J1 and J2 both fall within limits. In that case our best bet is probably to clip them as before as warn the user. I will think about it a bit and make changes soon

The other case that I will try to catch (and is not covered now) is that if a requested delta_theta is 0, but 0 violates the accel limits. It's not handled right now because of 1) a divide by 0, and 2) because linear scaling it won't help it get within bounds

Reply:
Alright, I messed around with this a lot and eventually decided to revert it to how it was before this PR. I might open an issue or a different PR on this topic later, but it was more difficult than I anticipated. I found that acceleration limits actually needed to be enforced a lot of the time, especially when starting or stopping jogging. During enforcing, linear scaling couldn't get all joints within limits a significant number (~80%) of the times, and the clipping was used anyway.

With the latest commit, the enforcement has not changed in this PR (except for a small bug). For now, I am accepting that the output will not be a scaled version of the input while we need to enforce accel limits